compatibility/bazel_tools/daml_script/daml_script.bzl

```diff
@@ -104,6 +104,9 @@ if [ {upload_dar} -eq 1 ] ; then
     --port 6865 \\
     $$(canonicalize_rlocation $(rootpath {dar}))
 fi
+
+sleep 2
```

Comment:
I'm not sure this is gonna fix the tests (concretely `//:daml-script-test-compiler-2.4.0-runner-0.0.0`) because the block above only runs when `{upload_dar} -eq 1`, which is only the case when `use_sandbox_on_x` - so not Canton

Reply:
Also, the `client_server_test` where this script is used as the client has `runner = "//bazel_tools/client_server:runner"`, which seems to wait for the port to be up, see below

https://github.com/digital-asset/daml/blob/553674a85024e23b789f12c2326c0b915de91a31/compatibility/bazel_tools/client_server/Main.hs#L20-L22

Reply:
perhaps this _is_ the fix if Canton is opening the port before connecting to the domain? That would explain the error we're getting,

```
Exception in thread "main" com.daml.lf.engine.script.ScriptF$FailedCmd: Command allocateParty failed: FAILED_PRECONDITION: PARTY_ALLOCATION_WITHOUT_CONNECTED_DOMAIN(9,2e397415): Cannot allocate a party without being connected to a domain
```

In that case, I wonder if there's a more principled way to check if canton is fully ready? I've found we sometimes use `client_server_test` with `runner = "//bazel_tools/client_server/runner_with_port_check"`, maybe switching this test to use that would work? cc @andreaslochbihler-da

Reply:
It is maybe not the good place, but it block 2s before running the script independently from "{upload_dar} -eq 1".

As far I understood @andreaslochbihler-da waiting for the port to be up is not enough.

Reply:
yes but maybe later it needs to be 3s and we end up debugging this again

Reply:
The only real fix is to retry to allocate party until it succeeds. 
This is a bit tricky in script, so I proposed to got with this "quick and dirty fix" to solve the annoying flakiness.

Reply:
I dug a bit and it seems #13463 fixed the same issue for other tests by switching the `client_server_test` runner from `@//bazel_tools/client_server/runner_with_port_check` to `@//bazel_tools/client_server/runner_with_health_check`. 
The difference is that the former just tries to connect to the port, while the latter actually checks that the `/health` endpoint on that port responds with a 200 code. The former (`runner_with_port_check`) is almost identical to the runner used here (`//bazel_tools/client_server:runner`).

Reply:
Oh, I remember this one. A plain `port check` may not be enough, but a health check should do.

Reply:
I experimented with using a copy of `//bazel_tools/client_server/runner_with_health_check` but I think I need to fix some the sandbox config to actually serve the healthcheck endpoint, I can take it over tomorrow if that's okay @remyhaemmerle-da. Feel free to merge this now though.